pkg/gardenlet/operation/botanist/operatingsystemconfig.go

```diff
@@ -57,6 +57,11 @@ func (b *Botanist) DefaultOperatingSystemConfig() (operatingsystemconfig.Interfa
 		valitailEnabled, valiIngressHost = true, b.ComputeValiHost()
 	}
 
+	kubeProxyEnabled := true
+	if b.Shoot.GetInfo().Spec.Kubernetes.KubeProxy != nil && b.Shoot.GetInfo().Spec.Kubernetes.KubeProxy.Enabled != nil {
+		kubeProxyEnabled = *b.Shoot.GetInfo().Spec.Kubernetes.KubeProxy.Enabled
+	}
```

Comment:
Cosmetic nit: It should be possible to reuse the already existing `KubeProxyEnabled` helper func: https://github.com/gardener/gardener/blob/d5ed70d4f21675eebc57de00788613f8bfc8446e/pkg/apis/core/v1beta1/helper/helper.go#L1019-L1022

I will try to include it in another PR, doesn't make sense to raise a PR for such small thing.